File: Matrix/matrix.cpp
#include <iostream>
#include <string>
 
class Matrix {
  private:
    int self_height;
    int self_width;

  public:
    Matrix(int height, int width) {
      self_height = height;
      self_width = width;
    }

    int getHeight() {
      std::cout <<  self_height;
      return self_height;
    }

    int getWidth() {
      std::cout << self_width;
      return self_width;
    }

    int generate() {
      int matrix[self_height][self_height];

      for(int i = 0; i < self_height; i++) {
        for(int j = 0; j < self_width; j++) {
            matrix[i][j] = j;
            std::cout << matrix[i][j] << " ";
        }
        std::cout << std::endl;
      }
      return 0;
    }
};
 
int main() {
  Matrix matrix(4,5);
  matrix.generate();
  return 0;
}

File: Matrix/matrix.go
package main

import "fmt"

type Matrix struct {
	height int
	width int
}

func (this *Matrix) generate() [][]int {
	matrix := make([][]int, this.height)
    for i := 0; i < this.height; i++ {
        for j := 0; j < this.width; j++ {
            matrix[i] = append(matrix[i], j)
        }
    }
    return matrix
}

func main() {
	matrix := Matrix{4,5}
	fmt.Println(matrix.generate())
}